dts/bindings/arm/nxp,bootpin.yaml

```diff
@@ -0,0 +1,17 @@
+# Copyright 2024, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  NXP Boot pins binding. This binding does not represent a full driver, but
+  declares pinctrl properties, so that the user can configure boot pins
+  at the board level.
+
+compatible: "nxp,bootpin"
```

Comment:
this looks like an incorrect usage of compatible because this is not an actual device/IP. Maybe `nxp-boot-pins`, like Arduino/ST morpho headers are specified.